### DISCUSSION TITLE
Swing-axle and beam-axle suspension behavior

### DISCUSSION THREAD

**Discussion by RenegadeScooter** [jrouwe/JoltPhysics#912] [Ideas]
I've noticed that some vehicles in GTA IV as well as Midnight Club have special suspension behavior, at least visually. Classic muscle cars, for example, have visible [swing-axle](https://en.wikipedia.org/wiki/Swing_axle) suspension for the front wheels and [beam-axle](https://en.wikipedia.org/wiki/Beam_axle) suspension for the rear wheels. Would it be possible to simulate these extra suspension behaviors, or am I just seeing an animation rig trick?

**Comment by jrouwe** [comment]
As a disclaimer: I'm certainly not an expert in racing games, but as far as I know, most of the time the visual representation of a vehicle is quite different from how it is simulated.

Swing-axle: My guess is that this may require some simulation component because when the wheels move independently it may be hard to represent it properly visually (although I don't know if it is possible to see the underside of the vehicle well enough to see that something is off). I realized that what I said on youtube will not work, the anti-rollbars (VehicleConstraintSettings::mAntiRollBars) are exactly the opposite of what you want. The code for making a swing-axle will however be very similar to it.

Beam-axle: I'm pretty certain that this is mostly a visualization thing as the difference with a linear suspension is minimal.
